solutions/usaco-1061.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-1061
+source: USACO Bronze 2020 December
+title: Stuck in a Rut
+author: Ryan Chou
+---
+
+[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_prob3_bronze_dec20.html)
+
+## Solution
+
+Since $x, y \leq 10^9$, we can't simulate each time period. Instead, try to identify when pairs of cows intersect. 
+
+One important observation is that cows can't move backwards, so the only way for two cows to collide is if $n[x] > e[x] $ and $n[y] < e[y]$. 
+
+This ensures that the cows are positioned like this:
+
+![](USACO-dec20-stuckinarut-bronze/intersection_example.png)
+
+Now we can sort cows by their static coordinates, which ensure that earlier collisions will execute first, and simulate those collisions.
+
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+<LanguageSection>
+<PySection>
+  
+```py
+n = int(input())
+ncows = []
+ecows = []
+
+for i in range(n):
+	dir, x, y = input().split()
+	# Also append i since we have to return them
+	# in the same order they were given to us.
+	if dir == 'N':
+		ncows.append((int(x), int(y), i))
+	else:
```

Comment:
honestly would prefer if you made this `elif dir == 'E':` but that i don't really care

Reply:
yeah i guess it would be better for beginners